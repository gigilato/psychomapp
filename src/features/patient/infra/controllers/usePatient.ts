import { useQuery } from '@tanstack/react-query'

import { getPatient } from '../fetchers'

export const getQueryKey = (id: string) => ['patients', id]
export const usePatient = (id: string) => {
  const query = useQuery({
    queryKey: getQueryKey(id),
    queryFn: async () => {
      return getPatient(id)
    },
  })
  return query
}
