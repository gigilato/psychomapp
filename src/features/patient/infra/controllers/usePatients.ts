import { useQuery } from '@tanstack/react-query'

import { getPatients } from '../fetchers'

export const queryKey = ['patients']
export const usePatients = () => {
  const query = useQuery({
    queryKey,
    queryFn: async () => {
      return getPatients()
    },
  })
  return query
}
