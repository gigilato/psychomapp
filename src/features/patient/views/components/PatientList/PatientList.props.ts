import { IFlatListProps } from '$organisms'
import { Patient } from '$types/database'

export interface IPatientListProps extends Omit<IFlatListProps<Patient>, 'data' | 'renderItem'> {}
