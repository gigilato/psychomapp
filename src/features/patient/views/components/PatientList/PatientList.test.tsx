import { render } from '@testing-library/react-native'

import { usePatients } from '$features/patient/infra/controllers/usePatients'
import { mockPatientList } from '$features/patient/infra/mocks/mockPatients'
import { QueryWrapper } from '$jest/jest.wrappers'

import { PatientList } from './PatientList'

jest.mock('$features/patient/infra/controllers/usePatients', () => ({
  usePatients: jest.fn(),
}))

describe('PatientList', () => {
  it('should render correctly', async () => {
    ;(usePatients as jest.Mock).mockImplementation(() => ({
      data: mockPatientList,
      isLoading: false,
      error: false,
    }))

    const Component = render(<PatientList />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })

  it('should render loading', async () => {
    ;(usePatients as jest.Mock).mockImplementation(() => ({
      data: undefined,
      isLoading: true,
      error: false,
    }))

    const Component = render(<PatientList />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })

  it('should render error', async () => {
    ;(usePatients as jest.Mock).mockImplementation(() => ({
      data: undefined,
      isLoading: false,
      error: true,
    }))

    const Component = render(<PatientList />, { wrapper: QueryWrapper })

    expect(Component).toMatchSnapshot()
  })
})
