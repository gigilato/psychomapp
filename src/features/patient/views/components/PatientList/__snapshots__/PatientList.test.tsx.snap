// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientList should render correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={
    <React.Fragment>
      <ErrorState
        className="mt-xxl"
        type="empty"
      />
    </React.Fragment>
  }
  ListFooterComponent={
    <React.Fragment>
      <Box
        className="h-m"
      />
    </React.Fragment>
  }
  ListHeaderComponent={
    <React.Fragment>
      <Box
        className="h-s"
      />
    </React.Fragment>
  }
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
      },
      undefined,
    ]
  }
  data={
    [
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "1",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "2",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "3",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "4",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "#FFFCF9",
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 16,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`PatientList should render error 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={
    <React.Fragment>
      <ErrorState
        className="mt-xxl"
        type="empty"
      />
    </React.Fragment>
  }
  ListFooterComponent={
    <React.Fragment>
      <Box
        className="h-m"
      />
    </React.Fragment>
  }
  ListHeaderComponent={
    <React.Fragment>
      <Box
        className="h-s"
      />
    </React.Fragment>
  }
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
      },
      undefined,
    ]
  }
  data={
    [
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "1",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "2",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "3",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "4",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "#FFFCF9",
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 16,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`PatientList should render loading 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListEmptyComponent={
    <React.Fragment>
      <ErrorState
        className="mt-xxl"
        type="empty"
      />
    </React.Fragment>
  }
  ListFooterComponent={
    <React.Fragment>
      <Box
        className="h-m"
      />
    </React.Fragment>
  }
  ListHeaderComponent={
    <React.Fragment>
      <Box
        className="h-s"
      />
    </React.Fragment>
  }
  contentContainerStyle={
    [
      {
        "flexGrow": 1,
      },
      undefined,
    ]
  }
  data={
    [
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "1",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "2",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "3",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
      {
        "birthdate": "2021-01-01",
        "createdAt": "",
        "firstname": "John",
        "id": "4",
        "lastname": "Doe",
        "practitionnerId": "",
        "updatedAt": "",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "#FFFCF9",
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <RCTRefreshControl />
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
      <View
        style={
          [
            {
              "height": 12,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#FFFCF9",
            },
            {
              "shadowColor": "rgba(0, 0, 0, 0.1)",
              "shadowOffset": {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 6,
            },
            {
              "shadowColor": "#D3D3D3",
            },
            {
              "marginLeft": 24,
              "marginRight": 24,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
            {
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
            },
            [
              {
                "opacity": undefined,
              },
              undefined,
            ],
          ]
        }
        testID=""
      >
        <Text
          style={
            [
              {
                "fontFamily": "SpaceGrotesk-Regular",
              },
              {
                "fontSize": 16,
              },
              {
                "lineHeight": 18,
              },
            ]
          }
        >
          John Doe
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          [
            {
              "height": 16,
            },
            [
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
